feat(tooltip): add tooltip component

brnrdog opened the pull request:
![tooltip](https://cloud.githubusercontent.com/assets/670325/16816726/af2edb00-4915-11e6-82f4-f4c6e4f82497.gif)

fernandofleury commented:
é possível centralizar o caret na tooltip? (centralizar também baseado no lado que ele abre)

brnrdog commented:
É, está mais para a direita. Vou dar uma olhada nisso, @fernandofleury

fernandofleury commented:
Se puder revisar o transition-out do tooltip

fernandofleury commented:
@brnrdog The doc is missing the `jQuery` usage

brnrdog commented:
done, @fernandofleury!

fernandofleury commented on src/css/atoms/tooltip.css:
```diff
@@ -0,0 +1,83 @@
+
+  .tooltip-arrow {
+    border-style: solid;
+    border-width: 7px;
```
Please use rem to refer to units.

fernandofleury commented on src/css/atoms/tooltip.css:
```diff
@@ -0,0 +1,83 @@
+    visibility: visible;
+  }
+
+  .tooltip-arrow {
```
I think this guy can move into an atom. And the positioning styles could go into the tooltip molecule

fernandofleury commented on src/css/atoms/tooltip.css:
```diff
@@ -0,0 +1,83 @@
+    margin-bottom: 10px;
+  }
+
+  &[x-placement^="top"] .tooltip-arrow {
```
These styles could move into tooltip molecule

fernandofleury commented on docs/js/tooltip.md:
```diff
@@ -0,0 +1,57 @@
+
+## How to
+
+To use the tooltip component, you need to add `[data-tooltip = 'tooltipSelector']` to the target element (you can use a class name, an id or a data attribute). The selector inserted should be present inside the target element, it'll be the tooltip itself. This component uses the [popper.js](http://popper.js.org/) library to deal with the positioning and jQuery.
```
I think you can spit the jQuery usage into the js usage section

fernandofleury commented:
@brnrdog Do not forget to include the `dist` folder into the final commit. They are the static files for package usage.

leandrogrillo commented:
@fernandofleury we already had problems including dist files when merging or rebasing to master.

I think it's better to add these files when create new tag.

leandrogrillo commented on src/css/atoms/tooltip.css:
```diff
@@ -0,0 +1,84 @@
+  transition: opacity .3s ease-in-out, visibility .6s;
+  visibility: hidden;
+
+  &:parent:hover > .tooltip {
```
is `:parent` a valid selector? 

If it's, which ie version is supported?

brnrdog replied:
Actually it isn't, my bad. I'm gonna fix it 😉

brnrdog commented:
@leandrogrillo should I remove the dist files?

fernandofleury commented:
@brnrdog  Fine by me

leandrogrillo commented:
@brnrdog @fernandofleury i prefer to remove the dist files and just add when create a new version of garden

:)

brnrdog commented:
@leandrogrillo I just deleted the dist files and rebased my branch if that's ok. If you prefer, I can keep the original dist files from the `master` branch or something.

leandrogrillo commented:
@brnrdog naaah, you don't need to do it. Just remove dist files and that's ok :)

leandrogrillo commented:
ty for the contribution @brnrdog
